ugvc/mrd/balanced_strand_utils.py

```diff
@@ -1218,23 +1224,15 @@ def convert_h5_to_papyrus_json(h5_file: str, output_json: str) -> str:
         h5_file_keys = store.keys()
     flatten_dfs = {}
 
-    # strand_ratio_category_counts
-    flatten_dfs["strand_ratio_category_counts"] = flatten_strand_ratio_category(
-        h5_file, "/strand_ratio_category_counts"
-    )
-
-    # strand_ratio_category_norm
-    flatten_dfs["strand_ratio_category_norm"] = flatten_strand_ratio_category(h5_file, "/strand_ratio_category_norm")
-
-    # stats_shortlist
-    flatten_dfs["stats_shortlist"] = flatten_metrics(h5_file, "/stats_shortlist")
-
-    # sorter stats
-    flatten_dfs["sorter_stats"] = flatten_metrics(h5_file, "/sorter_stats")
+    # flatten strand ratio categories
+    strand_ratio_to_convert = ["strand_ratio_category_counts", "strand_ratio_category_norm"]
+    flatten_dfs = {
+        key: flatten_strand_ratio_category(h5_file, key) for key in strand_ratio_to_convert if f"/{key}" in h5_file_keys
+    }
 
-    # category consensus
-    if "/df_category_concordance" in h5_file_keys:
-        flatten_dfs["category_consensus"] = flatten_metrics(h5_file, "/df_category_consensus")
+    # flatten 1D dataframes
+    keys_to_convert = ["stats_shortlist", "sorter_stats", "df_category_consensus"]
+    flatten_dfs = {key: flatten_metrics(h5_file, key) for key in keys_to_convert if f"/{key}" in h5_file_keys}
```

Comment:
There's a problem here - you are overriding the previous entries in flatten_dfs

Reply:
thanks XD